### PR TITLE
Make user.ext.prebid.buyeruids bidder name case insensitive

### PR DESCRIPTION
**Pull request by Sonali-More-Xandr**
We need to handle case insensitivity for user.ext.prebid.buyeruids biddernames. This PR updates the biddername look up to use the normalized names for buyeruids.BIDDER.

partially resolves https://github.com/prebid/prebid-server/issues/2400

### CODE REVIEW

File: endpoints/openrtb2/auction.go [hunk 1]
```diff
@@ -1782,7 +1782,9 @@ func (deps *endpointDeps) validateUser(req *openrtb_ext.RequestWrapper, aliases
 			return append(errL, errors.New(`request.user.ext.prebid requires a "buyeruids" property with at least one ID defined. If none exist, then request.user.ext.prebid should not be defined.`))
 		}
 		for bidderName := range prebid.BuyerUIDs {
-			if _, ok := deps.bidderMap[bidderName]; !ok {
+			normalizedCoreBidder, _ := openrtb_ext.NormalizeBidderName(bidderName)
+			coreBidder := normalizedCoreBidder.String()
+			if _, ok := deps.bidderMap[coreBidder]; !ok {
 				if _, ok := aliases[bidderName]; !ok {
 					return append(errL, fmt.Errorf("request.user.ext.%s is neither a known bidder name nor an alias in request.ext.prebid.aliases", bidderName))
 				}
```
Comment (gargcreation1992) [inline endpoints/openrtb2/auction.go:1790]: I think you could reuse `resolveBidder` function here. so - 
```
for bidderName := range prebid.BuyerUIDs {
       coreBidderName := resolveBidder(bidderName, aliases)
       if _, ok := deps.bidderMap[coreBidderName.String()]; !ok {
               return append(errL, fmt.Errorf("request.user.ext.%s is neither a known bidder name nor an alias in request.ext.prebid.aliases", bidderName))
       }
}
```

Since we already checked aliases so you need not to do it again. Let me know if this makes sense.

Comment (AlexBVolcy) [reply endpoints/openrtb2/auction.go:1790]: I implemented this change locally, and all of the tests still pass. You'd need to export the `resolveBidders` function since it comes from the exchange package.

File: endpoints/openrtb2/sample-requests/amp/valid-supplementary/aliased-buyeruids-case-insensitive.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (AlexBVolcy) [inline endpoints/openrtb2/sample-requests/amp/valid-supplementary/aliased-buyeruids-case-insensitive.json:1]: Could you also add a test using a bidder name where it's original formatting isn't already lowercase (i.e. yahooAds)

File: endpoints/openrtb2/auction.go (hunk 1 above)
Comment (Sonali-More-Xandr) [reply endpoints/openrtb2/auction.go:1790]: `resolveBidder` function is in the `exchange` package and auction.go file is in the `endpoints/openrtb2/`package. Exporting the function is an unsafe option so keeping it as is.

File: endpoints/openrtb2/sample-requests/amp/valid-supplementary/aliased-buyeruids-case-insensitive.json (hunk 2 above)
Comment (Sonali-More-Xandr) [reply endpoints/openrtb2/sample-requests/amp/valid-supplementary/aliased-buyeruids-case-insensitive.json:1]: Added in [8b7a55b](https://github.com/prebid/prebid-server/pull/3123/commits/8b7a55b3c066d5272be45bd29ca23ed3bf14fc75)

File: endpoints/openrtb2/auction.go [hunk 3]
```diff
@@ -1575,18 +1575,21 @@ func (deps *endpointDeps) parseBidExt(req *openrtb_ext.RequestWrapper) error {
...
 
 func (deps *endpointDeps) validateAliases(aliases map[string]string) error {
-	for alias, coreBidder := range aliases {
-		if _, isCoreBidderDisabled := deps.disabledBidders[coreBidder]; isCoreBidderDisabled {
-			return fmt.Errorf("request.ext.prebid.aliases.%s refers to disabled bidder: %s", alias, coreBidder)
+	for alias, bidderName := range aliases {
+		normalisedBidderName, _ := openrtb_ext.NormalizeBidderName(bidderName)
+		coreBidderName := normalisedBidderName.String()
+		if _, isCoreBidderDisabled := deps.disabledBidders[coreBidderName]; isCoreBidderDisabled {
+			return fmt.Errorf("request.ext.prebid.aliases.%s refers to disabled bidder: %s", alias, bidderName)
```
Comment (gargcreation1992) [inline endpoints/openrtb2/auction.go:1582]: Why was this changed? I think validateAliases were changed in - https://github.com/prebid/prebid-server/pull/3123/files#diff-23485c351c14cdca5e337d1c05105bd3a9b7308b017589654f0854ff4f9a7ff1 PR.

Comment (onkarvhanumante) [reply endpoints/openrtb2/auction.go:1582]: this happened because base PR PBS-1254-1 which  added changes for validateAliases was merged and deleted. Due to which commits from PBS-1254-1 got added in PBS-1257

**Comment by Sonali-More-Xandr** [comment]
Closing this PR since it has some commits from PBS-1254-1. Created a new PR with commits related to PBS-1257 only.
New PR - https://github.com/prebid/prebid-server/pull/3152
